c/include/foxglove-c/foxglove-c.h

```diff
@@ -45,6 +47,28 @@ typedef struct foxglove_server_options {
   const struct foxglove_server_callbacks *callbacks;
 } foxglove_server_options;
 
+typedef struct foxglove_mcap_options {
+  const char *path;
+  size_t path_len;
```

Comment:
@jtbandes What do you think about passing string data + size as two fields through the C API?

It's a little less natural if you have null-terminated strings (C) as you'd have to call strlen, but every other language using strings with a length. Even modern C usage might do that as well if they use a string library. This lets us use string_view for the C++ structs.

Reply:
I don't think I have a strong preference since I rarely work with C APIs. I can't say I have seen it much before, but it does have a concrete benefit. I guess we should also then change the other ones I've added to be consistent with this (could be a separate PR)

Reply:
Yeah, I think let's do that. You can create a ticket for it, and whoever gets to it first can do it